src/main/xerus/monstercat/downloader/SongView.kt

```diff
@@ -92,11 +88,58 @@ class SongView(private val sorter: ObservableValue<ReleaseSorting>):
 			}
 		}
 		
-		val defaultItems = {
-			arrayOf(MenuItem("Expand all") { expandAll() },
-				MenuItem("Collapse all") { expandAll(false) })
+		val menuPlay = MenuItem("Play") {
+			val selected = selectionModel.selectedItem ?: return@MenuItem
+			GlobalScope.launch {
+				Playlist.clear()
+				val value = selected.value
+				when(value) {
+					is Release -> Player.play(value)
+					is Track -> Player.playTrack(value)
+				}
+			}
+		}
+		val menuAdd = MenuItem("Add to playlist") {
+			val selected = selectionModel.selectedItem ?: return@MenuItem
+			GlobalScope.launch {
+				val value = selected.value
+				when(value) {
+					is Release -> {
+						value.tracks.forEach { track ->
+							Playlist.add(track)
+						}
+					}
+					is Track -> Playlist.add(value)
+				}
+			}
+		}
+		val menuAddNext = MenuItem("Play next") {
+			val selected = selectionModel.selectedItem ?: return@MenuItem
+			GlobalScope.launch {
+				val value = selected.value
+				when(value) {
+					is Release -> {
+						value.tracks.asReversed().forEach { track ->
+							Playlist.addNext(track)
+						}
+					}
+					is Track -> Playlist.addNext(value)
+				}
+			}
+		}
+		contextMenu = ContextMenu(
+				menuPlay, menuAdd, menuAddNext,
+				SeparatorMenuItem(),
+				MenuItem("Expand all") { expandAll() },
+				MenuItem("Collapse all") { expandAll(false) }
+		)
+		setOnContextMenuRequested {
+			val value = selectionModel.selectedItem.value
+			val enable = (value is Track || value is Release)
+			menuPlay.isDisable = !enable
```

Comment:
is there a reason you disable them instead of simply not displaying them?

Reply:
Makes for a more consistent UI. Users will know there is an option to expend all, or play.